src/Configuration.ts

```diff
@@ -0,0 +1,491 @@
+    const configuredArgs = yargs.config(config);
+    return configuredArgs.wrap(configuredArgs.terminalWidth()).parseSync(args);
+  }
+
+  configureOptions() {
+    const yargs = Yargs.usage(`Usage: ${this.programName} [options]`)
+      .example(
+        `${this.programName} -r ./src`,
+        "Running the tool with target directory 'src'"
+      )
+      .example(
+        `${this.programName} -r ./src --population_size 10`,
+        "Setting the population size"
+      )
+      .epilog("visit https://syntest.org for more documentation");
+
+    const generalOptions = this.configureGeneralOptions(yargs);
+    const fileOptions = this.configureFileOptions(generalOptions);
+    const dirOptions = this.configureDirectoryOptions(fileOptions);
+    const algorithmOptions = this.configureAlgorithmOptions(dirOptions);
+    const budgetOptions = this.configureBudgetOptions(algorithmOptions);
+    const loggingOptions = this.configureLoggingOptions(budgetOptions);
+    const postProcessingOptions =
+      this.configurePostProcessingOptions(loggingOptions);
+    const samplingOptions = this.configureSamplingOptions(
+      postProcessingOptions
+    );
+    const researchModeOptions =
+      this.configureResearchModeOptions(samplingOptions);
+
+    return researchModeOptions.showHidden(false);
+  }
+
+  configureGeneralOptions<T>(yargs: Yargs.Argv<T>) {
+    return (
+      yargs
+        // plugins
+        .option("plugins", {
+          alias: ["p"],
+          default: [],
+          description: "List of dependencies or paths to plugins to load",
+          group: "General options:",
+          hidden: false,
+          type: "array",
+        })
+        // ui
+        .option("user-interface", {
+          alias: [],
+          choices: ["regular"],
+          default: "regular",
+          description: "The user interface you use",
+          group: "General options:",
+          hidden: false,
+          type: "string",
+        })
+    );
+  }
+
+  configureFileOptions<T>(yargs: Yargs.Argv<T>) {
+    return (
+      yargs
+        // files
+        .option("target-root-directory", {
+          alias: ["r"],
+          demandOption: true,
+          description: "The root directory where all targets are in",
+          group: "File options:",
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+        .option("include", {
+          alias: ["i"],
+          default: ["./src/**/*.*"],
+          description: "Files/Directories to include",
+          group: "File options:",
+          hidden: false,
+          normalize: true,
+          type: "array",
+        })
+        .option("exclude", {
+          alias: ["e"],
+          default: [],
+          description: "Files/Directories to exclude",
+          group: "File options:",
+          hidden: false,
+          normalize: true,
+          type: "array",
+        })
+    );
+  }
+
+  configureDirectoryOptions<T>(yargs: Yargs.Argv<T>) {
```

Comment:
I would rename this to `configureOutputOptions`,  `configureOutputDirectoryOptions`, or `configureStorageOptions`